code-challenges.test.js

```diff
@@ -18,48 +20,111 @@
+//Once the test was written it was easier for me to figure out how to write the function in order to get a pass.
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-
-var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
-
-
-
-// b) Create the function that makes the test pass.
-
+// describe("capitalize the frist letter all the words in this array", () => {
+// var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+// var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
+// // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+// it("returns the words in an array with the first letter capitalized based on input", () => {
+//   expect(capitalize(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//   expect(capitalize(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+//   })
+// })
+// // b) Create the function that makes the test pass.
+
+// const capitalize = (array) => {
+//   var newArray = []
+//   for(let i=0; i<array.length; i++){
+//     newArray.push(array[i].charAt().toUpperCase())
+//     }
+//    return newArray
+//   }
+/////////////////////////////////////// Todd's Pseudo Code /////////////////////////////////////////////////////
+// This one has given me a super hard time right when I think I know how to do it it comes back around to slap me in the face
+// I feel like a new baby trying to learn how to swim and I just keep sucking in water.
+// I've bug Sarah so much that even with her excellent skills at teaching I still have no idea where I am
+//I understand the next step is a join but I have no idea where it goes I know that right now
+// The array is getting broken down and the two upper case is capitalizing every letter.
+// I feel terrible that I was unable to complete this question but I've already wasted five hours on this.
+// Already finished the other questions took a good night sleep and now I'm back this morning looking over my code.
+// But the test is giving me a different output than it was yesterday I don't know where I went wrong but I still have an hour before class so I can try and figure it out.
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
I feel ya man, this stuff is hard. Although I gotta say your logic as far as you got it is good! I think one of the things you're running into here is the problem of pseudo coding after you're done. You can use pseudo code to draw a map for yourself to follow if you want. We teach you this because sometimes it's harder to think about the problem while looking at code than it is while using plan language. How would you solve this with "plain language coding"? 
eg. step 4: separate the first letter in the  array and then run a string method .toUpperCase() on it...